app/src/main/java/com/codesoom/assignment/network/HttpRouter.java

```diff
@@ -0,0 +1,98 @@
+package com.codesoom.assignment.network;
+
+import com.sun.net.httpserver.HttpExchange;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * 등록된 path, method에 맞는 executor를 실행해줍니다
+ */
+public class HttpRouter {
+
+    HashMap<HttpRouterKey, RouterExecutable> pathMap = new HashMap<>();
+
+    public void get(String path, RouterExecutable executor) {
+        pathMap.put(new HttpRouterKey(HttpMethod.GET, path), executor);
+    }
+
+    public void post(String path, RouterExecutable executor) {
+        pathMap.put(new HttpRouterKey(HttpMethod.POST, path), executor);
+    }
+
+    public void put(String path, RouterExecutable executor) {
+        pathMap.put(new HttpRouterKey(HttpMethod.PUT, path), executor);
+    }
+
+    public void patch(String path, RouterExecutable executor) {
+        pathMap.put(new HttpRouterKey(HttpMethod.PATCH, path), executor);
+    }
+
+    public void delete(String path, RouterExecutable executor) {
+        pathMap.put(new HttpRouterKey(HttpMethod.DELETE, path), executor);
+    }
+
+    public void route(HttpExchange exchange) throws IOException {
+        final String method = exchange.getRequestMethod();
+        final HttpResponse responder = new HttpResponse(exchange);
+
+        if (method == null) {
+            responder.send(400, null);
+            return;
+        }
+
+        final URI uri = exchange.getRequestURI();
+        final String path = uri.getPath();
```

Comment:
```suggestion
        final String path = exchange.getRequestURI().getPath();
```
uri는 생성되자마자 사용되지 않는 이름이에요 바로 path를 가져오기 위한 다리일 뿐이죠!
코드가 마음에 들지 않는다면 함수를 이용해서 분리하는 것이 좋겠어요!